cmd/infra/powervs/create.go

```diff
@@ -0,0 +1,1170 @@
+	"github.com/IBM-Cloud/power-go-client/ibmpisession"
+	"github.com/IBM-Cloud/power-go-client/power/models"
+	"github.com/IBM/go-sdk-core/v5/core"
+	"github.com/IBM/networking-go-sdk/directlinkv1"
+	"github.com/IBM/networking-go-sdk/zonesv1"
+	"github.com/IBM/platform-services-go-sdk/globalcatalogv1"
+	"github.com/IBM/platform-services-go-sdk/iamidentityv1"
+	"github.com/IBM/platform-services-go-sdk/resourcecontrollerv2"
+	"github.com/IBM/platform-services-go-sdk/resourcemanagerv2"
+	"github.com/IBM/vpc-go-sdk/vpcv1"
+
+	hypershiftLog "github.com/openshift/hypershift/cmd/log"
+)
+
+var cloudApiKey = os.Getenv("IBMCLOUD_API_KEY")
+
+const (
+	// Resource name suffix for creation
+	cloudInstanceNameSuffix = "nodepool"
+	vpcNameSuffix           = "vpc"
+	vpcSubnetNameSuffix     = "vpc-subnet"
+	cloudConnNameSuffix     = "cc"
+
+	// Default cloud connection speed
+	defaultCloudConnSpeed = 5000
+
+	// CIS service name
+	cisService = "internet-svcs"
+
+	// PowerVS service and plan name
+	powerVSService     = "power-iaas"
+	powerVSServicePlan = "power-virtual-server-group"
+
+	// Resource desired states
+	vpcAvailableState               = "available"
+	cloudInstanceActiveState        = "active"
+	dhcpServiceActiveState          = "ACTIVE"
+	cloudConnectionEstablishedState = "established"
+
+	// Resource undesired state
+	dhcpServiceErrorState = "ERROR"
+
+	// Time duration for monitoring the resource readiness
+	pollingInterval                  = time.Second * 5
+	vpcCreationTimeout               = time.Minute * 5
+	cloudInstanceCreationTimeout     = time.Minute * 5
+	cloudConnEstablishedStateTimeout = time.Minute * 30
+	dhcpServerCreationTimeout        = time.Minute * 30
+	cloudConnUpdateTimeout           = time.Minute * 10
+
+	// Service Name
+	powerVsService  = "powervs"
+	vpcService      = "vpc"
+	platformService = "platform"
+)
+
+var customEpEnvNameMapping = map[string]string{
+	powerVsService:  "IBMCLOUD_POWER_API_ENDPOINT",
+	vpcService:      "IBMCLOUD_VPC_API_ENDPOINT",
+	platformService: "IBMCLOUD_PLATFORM_API_ENDPOINT"}
+
+// CreateInfraOptions command line options for setting up infra in IBM PowerVS cloud
+type CreateInfraOptions struct {
+	BaseDomain             string
+	ResourceGroup          string
+	InfraID                string
+	PowerVSRegion          string
+	PowerVSZone            string
+	PowerVSCloudInstanceID string
+	PowerVSCloudConnection string
+	VpcRegion              string
+	Vpc                    string
+	OutputFile             string
+	Debug                  bool
+}
+
+type TimeDuration struct {
+	time.Duration
+}
+
+var unsupportedPowerVSZones = []string{"wdc06"}
+
+var powerVsDefaultUrl = func(region string) string { return fmt.Sprintf("https://%s.power-iaas.cloud.ibm.com", region) }
+var vpcDefaultUrl = func(region string) string { return fmt.Sprintf("https://%s.iaas.cloud.ibm.com/v1", region) }
+
+var log = func(name string) logr.Logger { return hypershiftLog.Log.WithName(name) }
```

Comment:
why do we need this? why can't we use the declared log variable in "github.com/openshift/hypershift/cmd/log" as we do for the other platforms?

Reply:
We wanted to add Name in all the logs specific to an instance of infra creation or destroy. 
So added this to shorten `log.Log.WithName` to log